gherkin/perl/lib/Gherkin/Pickles/Compiler.pm

```diff
@@ -68,9 +77,17 @@ sub _compile_scenario {
                 $class->_create_pickle_arguments(
                     $step,
                     $variables, $values );
+
+            $last_keyword =
+                $inheriting_keyword{$step->keyword_type} ?
+                $last_keyword : $step->keyword_type;
+
+            #TODO: Throw an exception here when $last_keyword is empty
```

Comment:
Why throw an exception?

If last_keyword is empty, I would just consider it as the non specified keyword `*`

Reply:
`$last_keyword` is empty when there was no prior `Given`/`When`/`Then` keyword and the current keyword is `And` or `But`.

To be honest, I completely forgot about the non-specified keyword `*`. I think you're right that making `And` and `But` resolve to the non-specified keyword when there was no prior keyword specified, makes the most sense.